src/dmd/expressionsem.d

```diff
@@ -1926,6 +1928,17 @@ private bool functionParameters(const ref Loc loc, Scope* sc,
             }
             if (p.storageClass & STC.ref_)
             {
+                if (global.params.rvalueRefParam &&
+                    !arg.isLvalue() &&
+                    targ.isCopyable())
+                {   /* allow rvalues to be passed to ref parameters by copying
+                     * them to a temp, then pass the temp as the argument
+                     */
+                    auto v = copyToTemp(0, "__rvalue", arg);
```

Comment:
As far as I understand, this will generate a variable that is accessible from user code, if you know the name. Instead I recommend passing in `Identifier.anonymous` as the name (you need to add an overload to `copyToTemp`). This will set the variable declaration to be anonymous, which will not add it to the symbol table. This will prevent the symbol from being accessed by user code.

Reply:
Creating such temporaries is normal practice in D. Identifiers starting with a double _ are for the compiler only, but having them visible to debuggers, etc., is useful.

Reply:
> Creating such temporaries is normal practice in D

That doesn't mean it's good practice. There are a lot of things that are normal practice in D that are not good practice.

Reply:
>  Identifiers starting with a double _ are for the compiler only, but having them visible to debuggers, etc., is useful.

What's accessible in user code and to the debugger are two completely different things.